File: Flood fill Algorithm - GFG/flood-fill-algorithm.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
private:
    void dfs(int row,int col,vector<vector<int>> &image,vector<vector<int>> &ans,int newColor,int ini,int deltaRow[],int deltaCol[]){
        ans[row][col] = newColor;
        int n = image.size();
        int m = image[0].size();
        for(int i = 0;i<4;i++){
            int nRow = row + deltaRow[i];
            int nCol = col + deltaCol[i];
            if(nRow >= 0 && nRow < n && nCol >= 0 && nCol < m && image[nRow][nCol] == ini && ans[nRow][nCol] != newColor){
                dfs(nRow,nCol,image,ans,newColor,ini,deltaRow,deltaCol);
            }
        }
    }
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        
        // Code here 
        vector<vector<int>> ans = image;
        int iniColor = image[sr][sc];
        int deltaRow[] = {-1,0,1,0};
        int deltaCol[] = {0,1,0,-1};
        dfs(sr,sc,image,ans,newColor,iniColor,deltaRow,deltaCol);
        return ans;

    }
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n, m;
		cin >> n >> m;
		vector<vector<int>>image(n, vector<int>(m,0));
		for(int i = 0; i < n; i++){
			for(int j = 0; j < m; j++)
				cin >> image[i][j];
		}
		int sr, sc, newColor;
		cin >> sr >> sc >> newColor;
		Solution obj;
		vector<vector<int>> ans = obj.floodFill(image, sr, sc, newColor);
		for(auto i: ans){
			for(auto j: i)
				cout << j << " ";
			cout << "\n";
		}
	}
	return 0;
}
// } Driver Code Ends

File: Flood fill Algorithm - GFG/flood-fill-algorithm.py
class Solution:
	def floodFill(self, image, sr, sc, newColor):
		#Code here
	    iniColor = image[sr][sc]
	    ans = image
	    deltaRow = [-1,0,1,0]
	    deltaCol = [0,1,0,-1]
	    n = len(image)
	    m = len(image[0])
	    def dfs(row,col):
	        ans[row][col] = newColor
	        for i in range(4):
	            nRow = row + deltaRow[i]
	            nCol = col + deltaCol[i]
	            
	            if nRow >= 0 and nRow < n and nCol >= 0 and nCol < m and image[nRow][nCol] == iniColor and ans[nRow][nCol] != newColor:
	                dfs(nRow,nCol)
	    dfs(sr,sc)
	    return ans
	            
	        
	   

#{ 
 # Driver Code Starts
import sys
sys.setrecursionlimit(10**7)
if __name__ == '__main__':

	T=int(input())
	for i in range(T):
		n, m = input().split()
		n = int(n)
		m = int(m)
		image = []
		for _ in range(n):
			image.append(list(map(int, input().split())))
		sr, sc, newColor = input().split()
		sr = int(sr); sc = int(sc); newColor = int(newColor);
		obj = Solution()
		ans = obj.floodFill(image, sr, sc, newColor)
		for _ in ans:
			for __ in _:
				print(__, end = " ")
			print()
# } Driver Code Ends